src/router.rs

```diff
@@ -0,0 +1,129 @@
+use crate::{alcazar::AlcazarError, http_request::HttpError};
+use std::str::FromStr;
+
+#[derive(PartialEq, Copy, Clone)]
+pub enum MethodType {
+    POST,
+    GET,
+    PATCH,
+    DELETE,
+    CONNECT,
+    OPTIONS,
+    TRACE,
+    HEAD,
+}
+
+impl FromStr for MethodType {
+    type Err = AlcazarError;
+
+    fn from_str(method: &str) -> Result<MethodType, AlcazarError> {
+        match method {
+            "POST" => Ok(MethodType::POST),
+            "GET" => Ok(MethodType::GET),
+            "PATCH" => Ok(MethodType::PATCH),
+            "DELETE" => Ok(MethodType::DELETE),
+            "CONNECT" => Ok(MethodType::CONNECT),
+            "OPTIONS" => Ok(MethodType::OPTIONS),
+            "TRACE" => Ok(MethodType::TRACE),
+            "HEAD" => Ok(MethodType::HEAD),
+            _ => Err(AlcazarError::HttpError(HttpError::MethodNotImplemented)),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct Endpoint {
+    method: MethodType,
+}
+
+impl Default for Endpoint {
+    fn default() -> Self {
+        Self {
+            method: MethodType::GET,
+        }
+    }
+}
+
+impl Endpoint {
+    pub fn new() -> Self {
+        Endpoint::default()
+    }
+
+    pub fn add_method(mut self, method: MethodType) -> Self {
+        self.method = method;
+        self
+    }
+}
+
+#[derive(Clone)]
+pub struct Route {
+    path: String,
+    endpoint: Endpoint,
+}
+
+impl Default for Route {
+    fn default() -> Self {
+        Self {
+            path: "/".into(),
+            endpoint: Endpoint::default(),
+        }
+    }
+}
+
+impl Route {
+    pub fn new() -> Self {
+        Route::default()
+    }
+
+    pub fn add_path(mut self, path: String) -> Self {
+        self.path = path;
+        self
+    }
+
+    pub fn add_endpoint(mut self, endpoint: Endpoint) -> Self {
```

Comment:
`add_*` name doesn't really make sense, because it overwrites the `endpoint` / `path`.
For now, it's probably okay, but we should change it.

Reply:
Totally make sense!

Reply:
For the routing, naming as `endpoint(...)` is more than enough. In any case, developer will declare endpoints as the part of routing. Internally it's just adding the endpoint and that's it